src/scripts/_difference_by/_difference_by.gml

```diff
@@ -0,0 +1,59 @@
+/// @func _difference_by(array, [arrays], iteratee)
+/// @desc Like _difference(), except that it accepts iteratee which is invoked for each element of each array to generate the criterion by which they are compared.
+/// @param {Array} array The array to inspect
+/// @param {Array} [arrays] The arrays whose values are to be excluded
+/// @param {Script} iteratee The script invoked on each element to generate comparison criteria
+/// @returns {Array} The difference between the first and the remaining arrays
+/*
+@example
+// _floor(x)
+return floor(x);
+
+var arr0 = _array_of(0.5, 1, 2);
+var arr1 = _array_of(0);
+var arr2 = _array_of(0.1, 2.9);
+_difference_by(arr0, arr1, arr2, _floor);
+// => [1];
+*/
+
+if (argument_count == 0) return array_create(0);

```

Comment:
You can just return `[0]` instead of using `array_create`. GM:S2 GML has support for native array definitions.

Reply:
array_create(0) is an array of zero size, so return [] or array_create(0) are both fine I think.